frontend/src/app/shared/dialog/team-management/team-management.component.ts

```diff
@@ -0,0 +1,122 @@
+import { OrganisationService } from '../../services/organisation.service';
+import { Observable, of } from 'rxjs';
+import { TeamService } from '../../services/team.service';
+import { Team } from '../../types/model/Team';
+import { TeamMin } from '../../types/model/TeamMin';
+import { ConfirmDialogComponent } from '../confirm-dialog/confirm-dialog.component';
+import { CONFIRM_DIALOG_WIDTH } from '../../constantLibary';
+import { OrganisationState } from '../../types/enums/OrganisationState';
+
+@Component({
+  selector: 'app-team-management',
+  templateUrl: './team-management.component.html',
+  styleUrls: ['./team-management.component.scss'],
+})
+export class TeamManagementComponent implements OnInit {
+  teamForm = new FormGroup({
+    name: new FormControl<string>('', [Validators.required, Validators.minLength(2), Validators.maxLength(250)]),
+    organisations: new FormControl<Organisation[]>([], [Validators.required]),
+  });
+  organisations$: Observable<Organisation[]> = new Observable<Organisation[]>();
+  public hasInActiveOrganisations: boolean = false;
+  protected readonly formInputCheck = formInputCheck;
+  protected readonly errorMessages: any = errorMessages;
+
+  constructor(
+    public dialogRef: MatDialogRef<TeamManagementComponent>,
+    private dialog: MatDialog,
+    private organisationService: OrganisationService,
+    private teamService: TeamService,
+    @Inject(MAT_DIALOG_DATA) public data: { team: TeamMin },
+  ) {}
+
+  ngOnInit(): void {
+    this.organisations$ = this.organisationService.getOrganisations();
+    if (this.data) {
+      this.organisationService
+        .getOrganisationsByTeamId(this.data.team.id)
+        .subscribe((organisationsOfTeam: Organisation[]) => {
+          this.mergeOrganisations(organisationsOfTeam);
+          this.teamForm.setValue({
+            name: this.data.team.name,
+            organisations: organisationsOfTeam,
+          });
+          this.hasInActiveOrganisations =
+            organisationsOfTeam.filter((organisation) => organisation.state != OrganisationState.ACTIVE).length > 0;
+        });
+    }
+  }
+
+  saveTeam() {
+    if (!this.data) {
+      let newTeam: Team = { ...this.teamForm.value, activeObjectives: 0 } as Team;
+      this.teamService.createTeam(newTeam).subscribe((result) => {
+        this.dialogRef.close(result);
+      });
+    } else {
+      let updatedTeam: Team = { ...this.teamForm.value, id: this.data.team.id } as Team;
+      this.teamService.updateTeam(updatedTeam).subscribe((result) => {
+        this.dialogRef.close(result);
+      });
+    }
+  }
+
+  mergeOrganisations(organisationsOfTeam: Organisation[]) {
+    this.organisations$.subscribe((activeOrganisations) => {
+      organisationsOfTeam
+        .filter((organisation) => organisation.state == OrganisationState.INACTIVE)
+        .forEach((organisation) => activeOrganisations.push(organisation));
+      this.organisations$ = of(activeOrganisations);
```

Comment:
Du weist ein neues Observable der Variable `organisations$` zu. Was passiert mit den Subscribers, welche am alten Observable hören?
Wenn du  neue Werte über den Observable verbreiten möchtest:
`this.organisations$.next(activeOrganisations);`

Reply:
Es gibt keine .next() Methode auf dem Observable. Die Subscribers müssen wir nicht updaten, aussert der User ändert die Organisationen eines Teams, in diesem Fall laden wir die ganze Overview neu, somit auch diesen Komponenten, womit die Subscribers wieder neu zugeordnet werden.